### PR TITLE
Add --color option to kubectl cluster-info cmd

### PR DESCRIPTION
**Pull request by hypnoglow**
**What this PR does / why we need it**:

Currently, there is no way to disable colors in `kubectl cluster-info`
command output. This commit adds a new flag `--color` that solves
this, thus you can do `kubectl cluster-info --color=false`.

**Special notes for your reviewer**:

**Release note**:

```release-note
Add --color flag to kubectl cluster-info command
```

### CODE REVIEW

File: pkg/kubectl/cmd/clusterinfo.go [hunk 1]
```diff
@@ -103,7 +112,7 @@ func (o *ClusterInfoOptions) Run() error {
 		if err != nil {
 			return err
 		}
-		printService(o.Out, "Kubernetes master", o.Client.Host)
+		printService(o.Out, "Kubernetes master", o.Client.Host, !o.NoColor)
```
Comment (dixudx) [inline pkg/kubectl/cmd/clusterinfo.go]: `!o.NoColor` is equal to `o.Color`

That's why I'd prefer to use `o.Color` directly.

File: pkg/kubectl/cmd/clusterinfo.go [hunk 2]
```diff
@@ -61,15 +66,19 @@ func NewCmdClusterInfo(f cmdutil.Factory, ioStreams genericclioptions.IOStreams)
...
+		Short:                 i18n.T("Display cluster info"),
+		Long:                  longDescr,
+		Example:               clusterinfoExample,
 		Run: func(cmd *cobra.Command, args []string) {
 			cmdutil.CheckErr(o.Complete(f, cmd))
 			cmdutil.CheckErr(o.Run())
 		},
 	}
+
+	cmd.Flags().BoolVar(&o.NoColor, "no-color", false, "Disable colors in the output")
```
Comment (dixudx) [inline pkg/kubectl/cmd/clusterinfo.go]: I'd prefer to use `--color` directly.

Please bind the default value in L65. 

And update as below,

```
cmd.Flags().BoolVar(&o.Color, "color", o.Color, "enable colors in the output")
```

**Review by dixudx — COMMENTED** [review]
Nice add-on.

File: pkg/kubectl/cmd/clusterinfo.go (hunk 2 above)
Comment (hypnoglow) [reply pkg/kubectl/cmd/clusterinfo.go]: Yeah, `--color` is much better, thanks!

**Comment by hypnoglow** [comment]
Changed formatting in L70-L74 in accordance with gofmt.

File: pkg/kubectl/cmd/clusterinfo.go [hunk 3]
```diff
@@ -58,10 +63,12 @@ type ClusterInfoOptions struct {
 func NewCmdClusterInfo(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
 	o := &ClusterInfoOptions{
 		IOStreams: ioStreams,
+		Color:     true,
 	}
 
 	cmd := &cobra.Command{
-		Use:     "cluster-info",
+		Use: "cluster-info",
+		DisableFlagsInUseLine: true,
```
Comment (mengqiy) [inline pkg/kubectl/cmd/clusterinfo.go]: Why you need this?

Comment (hypnoglow) [reply pkg/kubectl/cmd/clusterinfo.go]: I have no reason. I saw this widely used in other commands. I can remove this if we don't want this.

**Comment by hypnoglow** [comment]
Rebased, as it was a bit stale

**Pull request closed without merging**